src/pages/index.js

```diff
@@ -83,25 +89,122 @@ function handlePicturePopup(name, link) {
   popupImage.open(name, link);
 }
 
-const popupNewCard = new PopupWithForm(newCard, (data) => {
-  createCard(data);
-  popupNewCard.reset();
-  newCardFormValidator.disableSubmitButton();
-});
+const popupDelete = new PopupConfirmation(popupConfirmDelete, handleDeletePicture);
+
+function handleDeleteButton(cardId, element) {
+  popupDelete.open(cardId, element);
+}
+
+function handleDeletePicture(cardId, element) {
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      element.remove();
+    })
+    .catch((error) => {
+      console.error('Error deleting card:', error);
+    });
+}
+
+function handleLikeButton(cardId, isLiked, btnLike) {
+  if (isLiked) {
+    api
+      .dislikeCard(cardId)
+      .then(() => {
+        btnLike.classList.remove('card__btn-like_active');
```

Comment:
Needs correction: you should implement the method for rendering likes in Card class and use it here